### PR TITLE
WIP: Be explicit about least privileged security context

### PR DESCRIPTION
**Pull request by phsiao**
This branch is a continuation of #1188, to continue the securityContext part of least privilege by making it explicit.

### CODE REVIEW

File: pkg/operator/api/api.go [hunk 1]
```diff
@@ -160,6 +160,8 @@ func (c *Cluster) makeDeployment() *extensions.Deployment {
 
 func (c *Cluster) apiContainer() v1.Container {
 
+	securityContext := k8sutil.BuildSecurityContext(false, 0, false, false)
```
Comment (kokhang) [inline pkg/operator/api/api.go]: Nit: i would add comments to these args for better readability.
For example 
```securityContext := k8sutil.BuildSecurityContext(false /* not privileged */, 0, false /* run as root */, false /* RW filesystem */)```

Comment (jbw976) [reply pkg/operator/api/api.go]: this is where C#'s "named arguments" are so useful (https://stackoverflow.com/a/20601083/467462).  i wish golang had something similar.

Comment (phsiao) [reply pkg/operator/api/api.go]: I miss that from python too.

Comment (kokhang) [reply pkg/operator/api/api.go]: @phsiao i dont mind this too much. That is why is said it was a nit :).
It just makes it a bit more clear to me what the arguments mean without having to go the function definition.

Comment (phsiao) [reply pkg/operator/api/api.go]: Simple change for the better, happy to do it.  :)

**Comment by phsiao** [comment]
Going to rebase/squash the commits.

**Comment by phsiao** [comment]
Some changes may cause similar problem in #1209 so it needs to be reworked.

**Comment by bassam** [comment]
@phsiao I recommend we close this one for now. There is work in progress to support a new security design here: https://github.com/rook/rook/blob/master/design/security-model.md

**Comment by phsiao** [comment]
Agreed.  Closing.

**Pull request closed without merging**